File: automation/config/appium_config.py
import os

class AppiumConfig:
    HOST = os.getenv("APPIUM_HOST", "127.0.0.1")
    PORT = int(os.getenv("APPIUM_PORT", "4723"))
    APPIUM_SERVER_URL = f"http://{HOST}:{PORT}/"
    
    BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    WORKSPACE_DIR = os.path.dirname(BASE_DIR)
    BASE_URL = os.getenv("BASE_URL", "https://InthaSindhu.github.io/imageauthenticity-pdd/")
    
    APK_PATH = os.getenv("APK_PATH", os.path.join(WORKSPACE_DIR, "ImageAuthenticity-debug.apk"))
    PACKAGE_NAME = "com.imageauth.verifier"
    MAIN_ACTIVITY = "com.imageauth.verifier.MainActivity"
    
    ANDROID_CAPABILITIES = {
        "platformName": "Android",
        "appium:automationName": "UiAutomator2",
        "appium:deviceName": os.getenv("ANDROID_DEVICE_NAME", "Android Emulator"),
        "appium:app": APK_PATH,
        "appium:appPackage": PACKAGE_NAME,
        "appium:appActivity": MAIN_ACTIVITY,
        "appium:noReset": False,
        "appium:fullReset": False,
        "appium:autoGrantPermissions": True,
        "appium:newCommandTimeout": 300,
        "appium:adbExecTimeout": 60000
    }
    
    REPORTS_DIR = os.path.join(WORKSPACE_DIR, "Test Results")
    EXCEL_REPORTS_DIR = os.path.join(REPORTS_DIR, "Excel")
    HTML_REPORTS_DIR = os.path.join(REPORTS_DIR, "HTML")
    JSON_REPORTS_DIR = os.path.join(REPORTS_DIR, "JSON")
    SCREENSHOTS_DIR = os.path.join(REPORTS_DIR, "Screenshots")
    LOGS_DIR = os.path.join(REPORTS_DIR, "Logs")
    SUMMARY_DIR = os.path.join(REPORTS_DIR, "Summary")

File: appium-mobile-e2e/config/appium-config.js
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export const appiumConfig = {
  hostname: process.env.APPIUM_HOST || '127.0.0.1',
  port: parseInt(process.env.APPIUM_PORT || '4723', 10),
  path: '/',
  capabilities: {
    platformName: 'Android',
    'appium:automationName': 'UiAutomator2',
    'appium:deviceName': process.env.ANDROID_DEVICE_NAME || 'Android Emulator',
    'appium:app': path.resolve(__dirname, '../../ImageAuthenticity-debug.apk'),
    'appium:appPackage': 'com.imageauth.verifier',
    'appium:appActivity': 'com.imageauth.verifier.MainActivity',
    'appium:noReset': false,
    'appium:fullReset': false,
    'appium:autoGrantPermissions': true,
    'appium:newCommandTimeout': 120
  },
  reportsDir: path.resolve(__dirname, '../reports'),
  reportFileName: 'Image_Authenticity_Android_Appium_Report.xlsx',
  baseUrl: process.env.TEST_BASE_URL || 'http://localhost:5173'
};
